app/assets/javascripts/admin/addon/routes/admin-site-settings.js

```diff
@@ -2,29 +2,41 @@ import { action } from "@ember/object";
 import { service } from "@ember/service";
 import DiscourseRoute from "discourse/routes/discourse";
 import { i18n } from "discourse-i18n";
+import SiteSettingFilter from "admin/lib/site-setting-filter";
 import SiteSetting from "admin/models/site-setting";
 
 export default class AdminSiteSettingsRoute extends DiscourseRoute {
   @service siteSettingChangeTracker;
 
   queryParams = {
-    filter: { replace: true },
+    filter: {
+      replace: true,
+      refreshModel: true,
+    },
+    onlyOverridden: {
+      replace: true,
+      refreshModel: true,
+    },
   };
 
+  _siteSettings = null;
+
   titleToken() {
     return i18n("admin.config.site_settings.title");
   }
 
-  model() {
-    return SiteSetting.findAll();
-  }
-
-  afterModel(siteSettings) {
-    const controller = this.controllerFor("adminSiteSettings");
-
-    if (!controller.get("visibleSiteSettings")) {
-      controller.set("visibleSiteSettings", siteSettings);
+  async model(params) {
+    if (!this._siteSettings) {
```

Comment:
We could consider clearing this when exiting the site settings area, but I'm not sure if that'd serve any use.